### PR TITLE
Rename a few modules, shorten names, and add documentation

### PR DESCRIPTION
**Pull request by Geod24**
Some parts of the consensus package didn't make sense to me.
I hope the first 2 commits are no-brainers, and the third one will make sense to everyone.
The last one is a renamed of a name I find annoying, but if people prefer to keep the long name, or find another shorter name, suggestions welcome!

### CODE REVIEW

**Comment by kchulj** [comment]
`UTXO` LGTM. Since it's a data structure of a UTXO..

File: source/agora/consensus/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+It contains the definitions of the data types that are exchanged and stored,
+as well as the state a full node must keep.
+
+This classification derives from framing the problem as [state machine replication](https://en.wikipedia.org/wiki/State_machine_replication).
+
+Following the state machine's definition, we associate the following components:
+- **States**: `ValidatorSet` & `UTXOSet` represent the node's view of the network;
+- **Inputs**: The content of the blocks (`Block`, `Enrollment`, `Transaction`);
+- **Outputs**: In our system, output is equal to state, so there's no distinction;
+- **Transition function**: Most of the mutation method in `state` (e.g. applying transactions to `UTXOSet`) and the ccontent of the `validation` package;
```
Comment (AndrejMitrovic) [inline source/agora/consensus/README.md]: method => methods
ccontent => content

Sorry doing a suggest is hard on mobile.

File: source/agora/consensus/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- **Output function**: Like outputs, we don't have a separate representation for this;
+- **Start**: This is simply the genesis block.
+
+Our **Inputs** also includes some specific items, such as `agora.consensus.data.Params`,
+which should be constant for the lifetime of the machine, but is used to tweak behavior in tests.
```
Comment (AndrejMitrovic) [inline source/agora/consensus/README.md:18]: I think we might also use it for protocol upgrades in the future?

**Review by AndrejMitrovic — APPROVED** [review]
Just a nit.

File: source/agora/consensus/README.md (hunk 2 above)
Comment (Geod24) [reply source/agora/consensus/README.md:18]: If we do we can update the docs :P

File: source/agora/consensus/README.md (hunk 1 above)
Comment (Geod24) [reply source/agora/consensus/README.md]: Thanks, fixed

**Comment by Geod24** [comment]
Linking issue should be fixed (but also shouldn't happen in the first place, LDC is being funky).

**Comment by Geod24** [comment]
Green

**Pull request merged**